Add warning for environment section regarding shell interpreter command execution timeout (#378)
hpcng/singularity-userdocs: environment_and_metadata.rst

   $ singularity run env.sif 
   Hello

.. warning::
   Singularity 3.6 uses an embedded shell interpreter to evaluate and setup container
   environments, therefore all commands executed from the ``%environment`` section have
   an execution timeout of **5 seconds** for Singularity 3.6 and a **1 minute** timeout since
   Singularity 3.7. While it is fine to source a script from there, it is not recommended

>    to use this section to do potentially long initialization stuff because it would

Reviewer: ```suggestion
   to use this section to run potentially long initialization tasks because this would
```
